File: src/main/java/no/nav/sosialhjelp/soknad/consumer/virusscan/VirusScanConfig.java
package no.nav.sosialhjelp.soknad.consumer.virusscan;


import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.client.WebClient;

import java.net.URI;

@Configuration
public class VirusScanConfig {

    private static final URI DEFAULT_CLAM_URI = URI.create("http://clamav.nais.svc.nais.local/scan");

    @Bean
    public VirusScanner virusScanner(WebClient webClient) {
        final var clamAvWebClient = webClient.mutate()
                .baseUrl(DEFAULT_CLAM_URI.toString())
                .build();

        return new ClamAvVirusScanner(DEFAULT_CLAM_URI, clamAvWebClient);
    }
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/vedlegg/virusscan/VirusScanConfig.kt
package no.nav.sosialhjelp.soknad.vedlegg.virusscan

import no.nav.sosialhjelp.soknad.app.client.config.unproxiedWebClientBuilder
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.function.client.WebClient

@Configuration
class VirusScanConfig(
    @Value("\${virusscan_enabled}") private val enabled: Boolean,
    @Value("\${clamav_url}") private val clamAvUrl: String,
    webClientBuilder: WebClient.Builder,
) {

    @Bean
    fun virusScanner(): VirusScanner {
        return VirusScanner(virusScannerWebClient, enabled)
    }

    private val virusScannerWebClient: WebClient = unproxiedWebClientBuilder(webClientBuilder).baseUrl(clamAvUrl).build()
}
